cypress/e2e/files/rename-files.cy.ts

```diff
@@ -0,0 +1,73 @@
+import { faker } from "@faker-js/faker";
+
+describe("Rename File Tests", () => {
+  const username = faker.internet.userName();
+  const password = faker.internet.password({ length: 8 });
+  const username1 = faker.internet.userName();
+  const password1 = faker.internet.password({ length: 8 });
+  const folderName = faker.system.commonFileName();
+  const randomNewFolderName = faker.system.commonFileName();
+
+  it("Register a user and rename (Duplicate Name)", () => {
+    // Register a test user
+    cy.visit("/register");
+    cy.get("input[name=username]").type(username);
+    cy.get("input[name=password]").type(password);
+    cy.get("input[name=confirmPassword]").type(password);
+    cy.get("button").contains("Submit").click();
+    cy.url().should("include", "/files");
+
+    // Create a folder
+    cy.get("button").contains("New Folder").click();
+    cy.get("input[aria-label='New folder name']").type(folderName);
+    cy.get("button").contains("Create folder").click();
+    cy.get("div").should("contain", folderName);
+
+    // Click the three dots to open the dropdown menu
+    cy.get("button[aria-label='Dropdown to delete, edit, share, or move file']").click();
+
+    // Click the "Edit" button to rename the folder
+    cy.get("button[aria-label='Edit']").click();
+
+    // Assert the rename modal is open
+    cy.get("div[role=dialog]").should("be.visible");
+    cy.get("div[role=dialog]").should("contain", "Enter a new name for the item");
+
+    // Click the "Save" button
+    cy.get("button").contains("Save").click();
+
+    cy.contains("A file with the same name already exists in the file directory").should("be.visible");
+  });
+
+  it("Rename folder", () => {
+    cy.visit("/register");
```

Comment:
It is not necessary to register another user or create another folder; you can use the ones you've used in the previous tests